src/VirtualBarometer.cpp

```diff
@@ -0,0 +1,22 @@
+#include "VirtualBarometer.h"
+
+VirtualBarometer::VirtualBarometer(unsigned int rs, double alt, double ns): 
+    random_seed{rs}, altitude_of_launch_site{alt}, noise{ns}, position_provider{},
+    rng {rs}, distribution {-ns, ns}
+{
+    // initialize position provider
+    current_position = Vector3{0, 0, altitude_of_launch_site};
+};
+
+void VirtualBarometer::process(double deltatime) {
+    std::cout << "PROCESSING NEW POSITION" << '\n';
+    position_provider.process(deltatime);
```

Comment:
Not sure if you folks are changing the architecture, but you probably want to call `position_provider.process(deltatime)` in the main simulation loop as a first step, then call `VirtualBarometer::process()` after it. This prevents coupling that is present here.